RESTserver/src/main/java/com/softserve/if072/restservice/controller/ProductController.java

```diff
@@ -1,43 +1,118 @@
     @ResponseBody
     @ResponseStatus(value = HttpStatus.OK)
-    public Product getById(@PathVariable int id){return productService.getById(id);}
+    public Product getProductById(@PathVariable int id, HttpServletResponse response) {
+        try {
+            Product product = productService.getProductById(id);
+            LOGGER.info(String.format("Product with id %d was retrieved", id));
+            return product;
+        } catch (DataNotFoundException e) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            LOGGER.error(String.format(productNotFound, id), e);
+            return null;
+        }
+    }
 
-    @RequestMapping(value = "/add", method = RequestMethod.POST)
-    @ResponseBody
+    @PostMapping(value = "/")
+    @ResponseStatus(value = HttpStatus.CREATED)
+    public void addProduct(@RequestBody Product product) {
+        productService.addProduct(product);
+        LOGGER.info("New product was created");
+    }
+
+    @PutMapping(value = "/")
     @ResponseStatus(value = HttpStatus.OK)
-    public void insert(@RequestBody Product product){productService.insert(product);}
+    public void update(@RequestBody Product product, HttpServletResponse response) {
+        int id = product.getId();
+        try {
+            productService.updateProduct(product);
+            LOGGER.info(String.format("Product with id %d was updated", id));
+        } catch (DataNotFoundException e) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            LOGGER.error(String.format("Cannot update Product with id %d", id), e);
+        }
+    }
 
-    @RequestMapping(value = "/update", method = RequestMethod.PUT)
-    @ResponseBody
+    @DeleteMapping(value = "/{id}")
     @ResponseStatus(value = HttpStatus.OK)
-    public void update(@RequestBody Product product){productService.update(product);}
+    public void delete(@PathVariable int id, HttpServletResponse response) {
+        try {
+            productService.deleteProduct(id);
+            LOGGER.info(String.format("Product with id %d was deleted", id));
+        } catch (DataNotFoundException e) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            LOGGER.error(String.format(productNotFound, id), e);
+        }
+    }
 
-    @RequestMapping(value = "/delete/{id}", method = RequestMethod.DELETE)
+    @GetMapping("/{id}/stores")
     @ResponseBody
     @ResponseStatus(value = HttpStatus.OK)
-    public void delete(@PathVariable int id){productService.delete(id);}
+    public List<Store> getAllStoresFromProduct(@PathVariable int id, HttpServletResponse response) {
+        try {
+            List<Store> stores = productService.getStoresByProductId(id);
+            LOGGER.info("All Stores were found");
+            return stores;
+        } catch (DataNotFoundException e) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            LOGGER.error("Stores were not found", e);
```

Comment:
No need to duplicate message. You can just use e.getMessage().
This is related to all the above methods